### PR TITLE
fix(asr): SubsamplingReductionModule pooling computes lengths for a single MaxPool1d pass

### LINKED ISSUE #16224: SubsamplingReductionModule pooling returns wrong lengths (factor 4 -> [6,5], factor 8 -> [0,0])
### Description

`SubsamplingReductionModule(reduction='pooling')` computes output lengths with `repeat_num=log2(reduction_factor)`, but the module applies a **single** `MaxPool1d(kernel_size=reduction_factor)` pass. The lengths are therefore wrong for every factor except 2:

- input lengths [100, 90], factor 4 → returns **[6, 5]** (actual downsampled T is 25/22)
- factor 8 → returns **[0, 0]** (actual 12/11)

With lengths 0, the conformer attention mask drops every frame — anything consuming `out_len` sees an empty sequence.

### Steps to reproduce

```python
import torch
from nemo.collections.asr.parts.submodules.subsampling import SubsamplingReductionModule

m = SubsamplingReductionModule(reduction='pooling', d_model=32, reduction_factor=8)
out, out_len = m(torch.randn(2, 32, 100), torch.tensor([100, 90]))
print(out_len)  # tensor([0, 0])
```

### Expected behavior

`[25, 22]` for factor 4, `[12, 11]` for factor 8 (one pooling pass divides T by the factor).

### Actual behavior

`calc_length(..., repeat_num=self._sampling_num)` with `_sampling_num = log2(reduction_factor)` — that repeat count is for the striding variant (which applies log2 strided convs), not the pooling one.

### Environment

NeMo main (`de26b36`), CPU, torch 2.14

### PR DESCRIPTION
**Pull request by ManoharPaturi**
Fixes #16224.

One line: the pooling variant applies a single `MaxPool1d(kernel_size=reduction_factor)`, so `calc_length` must use `repeat_num=1` — the `log2(reduction_factor)` repeat count belongs to the striding variant, which actually applies that many strided convs.

Regression test asserts lengths [25, 22] at factor 4 and [12, 11] at factor 8 (fails on main with [6, 5] / [0, 0]).

### CODE REVIEW

File: nemo/collections/asr/parts/submodules/subsampling.py [hunk 1]
```diff
@@ -699,7 +699,7 @@ def forward(self, x, lengths):
                 kernel_size=self.kernel_size,
                 stride=self.stride,
                 ceil_mode=False,
-                repeat_num=self._sampling_num,
+                repeat_num=1,  # a single MaxPool1d(kernel_size=reduction_factor) is applied below
```
Comment (nithinraok) [inline nemo/collections/asr/parts/submodules/subsampling.py:746]: I guess with this change, attribute is un-used. Could you remove the `self._sampling_num` attribute

**Review by nithinraok — CHANGES_REQUESTED** [review]

**Comment by ManoharPaturi** [comment]
@nithinraok the attribute removal you asked for is in the branch along with your main merge, and the checks that run are green. just checking if anything else is needed from my side for this one.

**Comment by nithinraok** [comment]
@ManoharPaturi Looks like one of the CI test failing, I restarted again to make sure.

**Review by nithinraok — APPROVED** [review]
LGTM

**Comment by ManoharPaturi** [comment]
@nithinraok thanks for the restart. the run on 87b063e is clean so far, build, linting, install checks and codeql all green with nothing failing. will keep an eye on the remaining queue.
